flink-docs/src/test/java/org/apache/flink/docs/rest/RestOptionsDocsCompletenessITCase.java

```diff
@@ -0,0 +1,281 @@
+            Assert.assertEquals(restAPI.method, documentedOption.method);
+            Assert.assertEquals(restAPI.statusCode, documentedOption.statusCode);
+            Assert.assertEquals(restAPI.methodDescription, documentedOption.methodDescription);
+
+            Map<String, String> expectedPathParameters =
+                    restAPI.pathParameters.pathParameters.stream()
+                            .collect(Collectors.toMap(t -> t.key, t -> t.description));
+            Map<String, String> actualPathParameters =
+                    documentedOption.pathParameters.stream()
+                            .collect(Collectors.toMap(t -> t.key, t -> t.description));
+            Assert.assertEquals(expectedPathParameters, actualPathParameters);
+
+            Map<String, Tuple2<Boolean, String>> expectedQueryParameters =
+                    restAPI.queryParameters.queryParameters.stream()
+                            .collect(
+                                    Collectors.toMap(
+                                            t -> t.key,
+                                            t -> Tuple2.of(t.mandatory, t.description)));
+            Map<String, Tuple2<Boolean, String>> actualQueryParameters =
+                    documentedOption.queryParameters.stream()
+                            .collect(
+                                    Collectors.toMap(
+                                            t -> t.key,
+                                            t -> Tuple2.of(t.isMandatory, t.description)));
+            Assert.assertEquals(expectedQueryParameters, actualQueryParameters);
+        }
+
+        Assert.assertTrue(documentedOptions.isEmpty());
+    }
+
+    private static final class DocumentedOption {
+        private final String url;
+        private final String method;
+        private final String statusCode;
+        private final String methodDescription;
+        private final Collection<RestAPIDocGenerator.PathParameter> pathParameters;
+        private final Collection<RestAPIDocGenerator.QueryParameter> queryParameters;
+
+        private DocumentedOption(
+                String url,
+                String method,
+                String statusCode,
+                String methodDescription,
+                Collection<RestAPIDocGenerator.PathParameter> pathParameters,
+                Collection<RestAPIDocGenerator.QueryParameter> queryParameters) {
+            this.url = url;
+            this.method = method;
+            this.statusCode = statusCode;
+            this.methodDescription = methodDescription;
+            this.pathParameters = pathParameters;
+            this.queryParameters = queryParameters;
+        }
+    }
+
+    private static RestAPISnapshot parseRestSnapshot(
+            String rootDir, String versionedSnapshotFileName) throws IOException {
+        final Path restSnapshotFolder =
+                Paths.get(rootDir, "flink-runtime-web", "src", "test", "resources")
```

Comment:
since the flink-runtime-web test-jar is on the classpath you could instead load this file as a resource instead.